src/Table/TableController.cc

```diff
@@ -202,7 +202,36 @@ void TableController::OnFileListRequest(
     file_list_response.set_parent(parent_path.string());
 
     try {
+        // get total number of files or directories under the directory
+        auto total_regular_files =
+            std::count_if(fs::directory_iterator(file_path), fs::directory_iterator{}, (bool (*)(const fs::path&))fs::is_regular_file);
+        auto total_directories =
+            std::count_if(fs::directory_iterator(file_path), fs::directory_iterator{}, (bool (*)(const fs::path&))fs::is_directory);
+        auto total_files = total_regular_files + total_directories;
+
+        // initialize variables for the progress report and the interruption option
```

Comment:
A lot of this code is duplicated between this function and the one in `FileListHandler.cc`. Would it make sense to factor out the common parts to a generic function which could be called by both?